### PR TITLE
refactor(core/utils): define TS ResourceHintOption for createResourceHint()

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Imma learning the types escripts!

File: src/geonovum/style.js [hunk 1]
```diff
@@ -73,12 +76,11 @@ function createResourceHints() {
...
-    .map(createResourceHint)
-    .reduce((frag, link) => {
-      frag.appendChild(link);
-      return frag;
-    }, document.createDocumentFragment());
+  ];
+  const resourceHints = opts.map(createResourceHint).reduce((frag, link) => {
+    frag.appendChild(link);
+    return frag;
+  }, document.createDocumentFragment());
```
Comment (sidvishnoi) [inline src/geonovum/style.js:83]: While we're at it, can we remove the `reduce` and use a regular old `for .. of`?

File: src/geonovum/style.js [hunk 2]
```diff
@@ -49,7 +49,10 @@ function createStyle(css_name) {
 }
 
 function createResourceHints() {
-  const resourceHints = [
+  /**
+   * @type ResourceHintOption[]
+   * */
```
Comment (sidvishnoi) [inline src/geonovum/style.js:54]: Nit: I like these in one-line when we can :smile: 
```suggestion
  /** @type ResourceHintOption[] */
```

File: src/geonovum/style.js [hunk 3]
```diff
@@ -73,12 +74,11 @@ function createResourceHints() {
...
-    .map(createResourceHint)
-    .reduce((frag, link) => {
-      frag.appendChild(link);
-      return frag;
-    }, document.createDocumentFragment());
+  ];
+  const resourceHints = opts.map(createResourceHint).reduce((frag, link) => {
+    frag.appendChild(link);
+    return frag;
+  }, document.createDocumentFragment());
```
Comment (marcoscaceres) [inline src/geonovum/style.js:81]:
```suggestion
  const resourceHints = document.createDocumentFragment();
  for (const link of opts.map(createResourceHint){
    resourceHints.appendChild(link);
  };
```

File: src/w3c/style.js [hunk 4]
```diff
@@ -86,12 +89,11 @@ function createResourceHints() {
...
-    .map(createResourceHint)
-    .reduce((frag, link) => {
-      frag.appendChild(link);
-      return frag;
-    }, document.createDocumentFragment());
+  ];
+  const resourceHints = opts.map(createResourceHint).reduce((frag, link) => {
+    frag.appendChild(link);
+    return frag;
+  }, document.createDocumentFragment());
```
Comment (marcoscaceres) [inline src/w3c/style.js:96]:
```suggestion
  const resourceHints = document.createDocumentFragment();
  for (const link of opts.map(createResourceHint){
    resourceHints.appendChild(link);
  };
```

File: src/core/utils.js [hunk 5]
```diff
@@ -41,54 +41,18 @@ export const ISODate = new Intl.DateTimeFormat(["en-ca-iso8601"], {
...
-  "report",
-  "script",
-  "serviceworker",
-  "sharedworker",
-  "style",
-  "worker",
-  "xslt",
-  "",
-]);
-
```
Comment (sidvishnoi) [inline src/core/utils.js:68]: I like this! Build-time checks, and less bytes in production!

File: src/dini/style.js [hunk 6]
```diff
@@ -51,7 +51,10 @@ function createBaseStyle() {
 }
 
 function createResourceHints() {
-  const resourceHints = [
+  /**
+   * @type ResourceHintOption[]
+   */
```
Comment (sidvishnoi) [inline src/dini/style.js:56]:
```suggestion
  /** @type ResourceHintOption[]  */
```

File: src/w3c/style.js [hunk 7]
```diff
@@ -66,7 +66,10 @@ function selectStyleVersion(styleVersion) {
 }
 
 function createResourceHints() {
-  const resourceHints = [
+  /**
+   * @type ResourceHintOption[]
+   */
```
Comment (sidvishnoi) [inline src/w3c/style.js:71]:
```suggestion
  /** @type ResourceHintOption[]  */
```

File: src/dini/style.js [hunk 8]
```diff
@@ -66,12 +69,11 @@ function createResourceHints() {
...
-    .map(createResourceHint)
-    .reduce((frag, link) => {
-      frag.appendChild(link);
-      return frag;
-    }, document.createDocumentFragment());
+  ];
+  const resourceHints = opts.map(createResourceHint).reduce((frag, link) => {
+    frag.appendChild(link);
+    return frag;
+  }, document.createDocumentFragment());
```
Comment (sidvishnoi) [inline src/dini/style.js:76]:
```suggestion
  const resourceHints = document.createDocumentFragment();
  for (const link of opts.map(createResourceHint){
    resourceHints.appendChild(link);
  };
```

**Review by sidvishnoi — APPROVED** [review]
:shipit: 
Maybe edit the commit message to make it appear in single line, if possible.

File: src/type-helper.d.ts [hunk 9]
```diff
@@ -122,6 +122,29 @@ interface Conf {
   shortName: string;
 }
 
+type ResourceHintOption = {
+  /**
+   * The type of hint.
+   **/
+  hint?: "preconnect" | "preload" | "dns-prefetch" | "prerender";
```
Comment (sidvishnoi) [inline src/type-helper.d.ts:129]:
```suggestion
  hint: "preconnect" | "preload" | "dns-prefetch" | "prerender";
```

**Pull request merged**